### ISSUE TITLE
NumericPicker ignores Value the first time it is created

### ISSUE THREAD

**Issue by WayneHiller** [Megabit/Blazorise#4075]
Any page that has NumericPicker's will ignore the Value of a bound property the first time it is created (all will have zeros).  Any subsequent times that any page that is displayed with NumericPicker's it works correctly. It seems to be something global in the NumberPicker.

 ```
<NumericPicker Decimals="0" SelectAllOnFocus="true" @bind-Value="@entity.HeaderCount" />
```

.Net 6, Blazorise 1.6, Blazor Wasm. Chrome.

**Comment by stsrki** [comment]
Unable to reproduce:

**Code:**

```razor
<NumericPicker Decimals="0" SelectAllOnFocus @bind-Value="@HeaderCount" />

@code {
    decimal HeaderCount = 50;
}
```

**Open page:**

![image](https://user-images.githubusercontent.com/900302/189904673-f6d5fdca-b477-4457-be53-ae455b81e450.png)

---

Maybe you're loading the value at a different page rendering state?

PS. Closing now.

**Comment by WayneHiller** [comment]
The value is being assigned from an object property returned from an API call.

**Comment by WayneHiller** [comment]
I have narrowed it down to this in a standard .Net 6 Blazor Wasm with API project:
Note I am using Flurl for the API call.

Client Index.razor
```
<div>
    <NumericPicker Decimals="0" 
                    SelectAllOnFocus="true" 
                    @bind-Value="@Forecast.TemperatureC" />
</div>

@code {
    private WeatherForecast Forecast = new();

    protected override async Task OnAfterRenderAsync(bool firstRender)
    {
        await base.OnAfterRenderAsync(firstRender);

        Forecast = await NavMgr.BaseUri.AppendPathSegment("weatherforecast/getforecast").GetJsonAsync<WeatherForecast>();
        await InvokeAsync(StateHasChanged);
    }
}
```

Server WeatherForecastController.cs
```
        [HttpGet]
        [Route("getforecast")]
        public async Task<WeatherForecast> GetForecast()
        {
            return new WeatherForecast
            {
                Date = DateTime.Now,
                TemperatureC = 45,
                Summary = "Testing"
            };
        }
```

The first time you visit the Index page the NumericPicker will stay at 0. After that you can go to any page and back to Index and it will contain 45.

**Comment by WayneHiller** [comment]
Definitely is a timing issue with the StateHasChanged call. Adding a Delay to the API call fixes the issue.
```
        [HttpGet]
        [Route("getforecast")]
        public async Task<WeatherForecast> GetForecast()
        {
            await Task.Delay(500);

            return new WeatherForecast
            {
                Date = DateTime.Now,
                TemperatureC = 45,
                Summary = "Testing"
            };
        }
```

**Comment by WayneHiller** [comment]
@stsrki Will you be taking a look at this at all? If not I can clone the repo and debug to try an figure out what is happening.

**Comment by David-Moreira** [comment]
Might be a race condition then? I guess `NumericPicker` uses js behind the scnes, might have to do with it? 
@WayneHiller from all your testing do you happen to have a repro ready for us? That will help.

In the meantime I guess you found your temporary workaround. We'll take a look again as soon as possible, but we do have a lot on our hands currently.
Reopening issue. Please provide the repro if you have it, really helps! thanks!

**Comment by stsrki** [comment]
@WayneHiller we're at the moment trying to finish 1.1 so this is put on hold for a little. If you have some time to look into it we'll really appreciate it.

**Comment by WayneHiller** [comment]
I will have to take a look at it this weekend I don't really have a work around because I do not want to delay displaying every edit page for a second. I do have a test project that shows the issue, will create a repo.

**Comment by WayneHiller** [comment]
Is there any docs on how to run the E2E tests? I think that would be the best place to verify what is happening.

**Comment by WayneHiller** [comment]
The issue seems to be related to the amount of time it takes to load and setup the JSNumericPickerModule the first time it is used (called from NumericPicker.OnFirstAfterRenderAsync).  A work around is to just add a hidden NumericPicker to a global component like the menu etc. This will force the load of the JSNumericPickerModule and it will be cached.

**Comment by David-Moreira** [comment]
Right, it was our suspicion that it could be a race condition like that. 
I guess that's a decent workaround for now. 

Thanks for helping us troubleshoot it. We'll let you know once we take a look at it.

**Comment by stsrki** [comment]
Probably can be solved the same as #4316
